PracticeProject/API/API.swift

```diff
@@ -5,26 +5,17 @@
 
 import Foundation
 
+/// It is API protocol used for common Api handling
 protocol API {
-    init()
-
-    func fetchUsers(excludingUserWithID: String?, success: (UsersList) -> Void, failure: (FetchError) -> Void)
+    
+    /// Getting the response from user API's
+    /// - Parameters:
+    ///   - excludingUserWithID: It is exclude the user from user list
+    ///   - success: get the user list response
+    ///   - failure: get the error message
+    func fetchUsers(excludingUserWithID: String?, success: @escaping (UsersList?) -> Void, failure: @escaping (FetchError?) -> Void)
 }
 
-extension API {
-//    static func make() -> API {
-//        // TODO: Return the concrete implementation of API
-//    }
-}
+typealias UsersList = [UserModel]
 
-// TODO: Create a data type representing users-list (according to expected JSON response)
-// (See the JSON response at: https://jsonplaceholder.typicode.com/users)
-//
-typealias UsersList = Any
-
-// TODO (Bonus): Create a more specific error type.
-// This can help identify the nature of a particular failure case.
-// e.g. network timeout, badly formatted request or failing to decode/deserialize
-// a response could cause failure in a network request.
-//
-typealias FetchError = Any
+typealias FetchError = String
```

Comment:
Can you design an error object that best fits potential failures from the `fetch` API request? 
String is not a fitting error object and ties the client (caller) to a single handling which is to show an error message.
e.g. If the cause of the failure is network timeout, client might want to retry before showing any error. 
You do not have write a retry request handling but design the error object so it communicates the cause of the failure thoroughly.

Reply:
As per your suggestion, I have changed the Error object. Kindly check and let me know.

Reply:
See Apple's documentation and other sources to improve your error object. 
https://developer.apple.com/documentation/swift/error

The goal is to design it in such a way that the receiver / handler of the error gets explicit information on what the cause of the error is and can handle different error cases in different ways. Returning a "message" that is a String does not give the handler any chance but to display something to the user. Read more about custom error types to understand the concept.